Cabal-tests/tests/PrinterTests.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE CPP #-}
+module Main
+    ( main
+    ) where
+
+import Prelude ()
+import Prelude.Compat
+
+import Distribution.Types.GenericPackageDescription
+import Data.Maybe(catMaybes)
+import Test.Tasty
+import Data.Text(unpack)
+import Test.Tasty.HUnit
+
+import Control.Monad                               (unless)
+import Distribution.Fields                         (runParseResult)
+import Distribution.PackageDescription.Parsec      (parseGenericPackageDescription)
+import System.Directory                            (setCurrentDirectory)
+import System.Environment                          (getArgs, withArgs)
+import System.FilePath                             ((</>))
+import Data.Text.Encoding(decodeUtf8, encodeUtf8)
+import Distribution.PackageDescription.ExactPrint(exactPrint)
+import Data.TreeDiff
+import Text.PrettyPrint hiding ((<>))
+import Data.TreeDiff.QuickCheck (ediffEq)
+
+import qualified Data.ByteString       as BS
+
+tests :: TestTree
+tests = testGroup "printer tests"
+    [ printExact
+    ]
+
+-------------------------------------------------------------------------------
+-- Warnings
+-------------------------------------------------------------------------------
+
+-- Parse some cabal file - print it like cabal file
+printExact :: TestTree
+printExact = testGroup "printExact"
+    [
+      testParsePrintExact "bounded.cabal"
+    , testParsePrintExact "two-sections.cabal"
```

Comment:
@andreabedini this tests starts failing if I add your commiits, do you know why?